worth2/main/tests/test_views.py

```diff
@@ -12,42 +12,52 @@
 from worth2.main.models import Participant
 
 
-class AvatarSelectorTest(LoggedInParticipantTestMixin, TestCase):
+class AvatarSelectorBlockTest(LoggedInParticipantTestMixin, TestCase):
     def setUp(self):
-        super(AvatarSelectorTest, self).setUp()
+        super(AvatarSelectorBlockTest, self).setUp()
+
         self.avatar1 = AvatarFactory()
         self.avatar2 = AvatarFactory()
         self.avatar3 = AvatarFactory()
 
-    def test_get(self):
-        r = self.client.get(reverse('avatar-selector'))
-        self.assertEqual(r.status_code, 200)
-
-    def test_participant_with_no_avatar(self):
-        h = get_hierarchy('main', '/pages/')
-        root = h.get_root()
-        root.add_child_section_from_dict({
-            'label': 'Section 1',
-            'slug': 'section-1',
-            'pageblocks': [],
+        self.h = get_hierarchy('main', '/pages/')
+        self.root = self.h.get_root()
+        self.root.add_child_section_from_dict({
+            'label': 'Avatar Selector Section',
+            'slug': 'avatar-selector-section',
+            'pageblocks': [{
+                'block_type': 'avatar selector block',
+            }],
             'children': [],
         })
-        r = self.client.get('/pages/section-1')
+        self.url = '/pages/avatar-selector-section/'
 
-        # Assert that a participant with no avatar is redirected to the
-        # avatar selector when attempting to navigate to pagetree
-        self.assertEqual(r.status_code, 302)
-        self.assertRedirects(r, reverse('avatar-selector'))
+    def test_get(self):
+        r = self.client.get(self.url)
+        self.assertEqual(r.status_code, 200)
+        self.assertContains(r, 'Avatar Selector Section')
+
+        # TODO: having trouble here -- why isn't this pageblock
+        # displaying correctly in the tests?
+        #
+        # self.assertTemplateUsed(r, 'main/avatar_selector_block.html')
```

Comment:
I'm pretty sure the block template fragment will not appear in the "template used" list as it is being rendered by hand. See https://github.com/ccnmtl/django-pagetree/blob/master/pagetree/templatetags/render.py.